torch/csrc/autograd/python_variable.cpp

```diff
@@ -2247,6 +2250,65 @@ void ConcretePyInterpreterVTable::dispatch(
       op, stack, py::reinterpret_steal<py::object>(obj), "__torch_dispatch__");
 }
 
+void ConcretePyInterpreterVTable::python_dispatcher(
+    const c10::OperatorHandle& op,
+    c10::DispatchKeySet ks,
+    torch::jit::Stack* stack) const {
+  const auto& schema = op.schema();
+  const auto num_arguments = schema.arguments().size();
+  auto arguments = torch::jit::pop(*stack, num_arguments);
+
+  // Parse the name into namespace and name (no overload_name)
+  // TODO: put this into the library
+  const auto& qualified_name = op.operator_name().name;
+  const auto& overload_name = schema.overload_name();
+  auto pos = qualified_name.find("::");
+  TORCH_INTERNAL_ASSERT(pos != std::string::npos, qualified_name);
+  // Make me some null terminated strings
+  std::string ns_str = qualified_name.substr(0, pos);
+  const char* ns = ns_str.c_str();
+  const char* func_name = qualified_name.c_str() + pos + strlen("::");
+
+  // The plan: convert all the arguments back into PyObjects,
+  // extracting out the tensor handles, then call
+  // handle_torch_function_no_python_arg_parser
+  // NB: at the point arguments are pushed to the stack, ALL defaults
+  // are already present
+
+  py::gil_scoped_acquire g;
+
+  std::vector<py::handle> overloaded_args;
+  py::handle torch_api_function =
+      py::module::import("torch").attr("ops").attr(ns).attr(func_name);
+  py::handle torch_api_function_overload;
+  if (overload_name == "") {
+    torch_api_function_overload = torch_api_function.attr("default");
+  } else {
+    torch_api_function_overload =
+        torch_api_function.attr(overload_name.c_str());
+  }
+  std::string module_name_str = "torch.ops." + ns_str;
+
+  auto args_kwargs = parseIValuesToPyArgsKwargs(op, arguments);
+  auto args = std::move(args_kwargs.first);
+  auto kwargs = std::move(args_kwargs.second);
+
+  auto state = c10::impl::PythonDispatcherTLS::get_state().ptr(getPyInterpreter());
```

Comment:
iiuc we're getting a pointer to the pyobj representing the 'global python dispatcher', and we're verifying that it was created with the same py interpreter we're on now.  maybe s/state/py_dispatcher/ would make this a little more readable

Reply:
yes